python/pyiceberg/avro/file.py

```diff
@@ -173,7 +173,7 @@ def __enter__(self) -> AvroFile[D]:
             A generator returning the AvroStructs.
         """
         self.input_stream = self.input_file.open(seekable=False)
-        self.decoder = BinaryDecoder(self.input_stream)
+        self.decoder = StreamingBinaryDecoder(self.input_stream)
```

Comment:
Just to echo my thoughts here. Ideally, we want to read the whole file at once, instead of streaming through it, and potentially doing multiple calls to S3. Maybe we want to increase the buffer size? PyArrow supports `.read(None)` to read the whole file. Idk and probably needs some more investigation.